nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -75,6 +79,45 @@ in
           '';
       };
 
+      enableCryptoStability = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Use ZFS with latest crypto stability patches. This was merged into
+          zfs master on feb 2, 2018. Those stability patches features a format
+          change for encrypted datasets. Old native encrypted datasets can still
+          be accessed but as read-only.
+          See https://github.com/zfsonlinux/zfs/pull/6864 for more details.
+
+          If you have not created the root dataset as encrypted dataset but
+          used a childset as encryption root like the Nixos Wiki suggested ->
+          zroot/root/nixos (where 'zroot' is not encrypted but 'root' is encryp-
+          ted and nixos is encrypted as well due to inheritance), then updating
+          to the new format is a simlpe matter - you just need to have the free
+          space to do so:
+          0. Create a snapshot:
+                zfs snapshot zpool/root/nixos@now
+          1. Create a custom Nixos iso with crypto stability patch enabled
+          2. Boot into that live environment
+          3. Import the pool and load the key
+          4. Create a new encrypted dataset, e.g.
+                zfs create -o encryption=aes-256-gcm -o keyformat=passphrase
+                -o mountpoint=none zroot/rootNEW
+          5. Use zfs send and receive to copy the data to new format:
+                zfs send zpool/root/nixos@now | zfs receive zpool/rootNew/nixos
+          6. Set correct mountpoint for the newly created dataset:
+                zfs set moutpoint=legacy zpool/root/New/nixos
+          7. Rename the old dataset and the new one:
+                zfs rename zpool/root zpool/rootOLD
+                zfs rename zpool/rootNEW zpool/root
+          8. That should allow to boot Nixos already with new format. If you
+          other encrypted mounts, you will probably need to convert them to new
+          format as well first.
+          If you have encrypted your root dataset you will need to nuke it
+          and re-create completely anew.
```

Comment:
IMO, this should _not_ be in the option description. The option description should specify what it does and the caveat about backwards compatibility, but then link elsewhere for migration details.

Reply:
I propose to invert option instead to help users to migrate: https://github.com/NixOS/nixpkgs/pull/34559#discussion_r165828101